indexer/feature_data.cpp

```diff
@@ -262,19 +263,21 @@ bool FeatureParams::AddHouseName(string const & s)
   return false;
 }
 
-bool FeatureParams::AddHouseNumber(string const & ss)
+bool FeatureParams::AddHouseNumber(string houseNumber)
 {
-  if (!feature::IsHouseNumber(ss))
+  // Remove trailing zero's from house numbers,
+  // it's important for debug checks of serialized-deserialized feature.
+  // Also replace full-width digits, mostly in Japan, by ascii-ones.
+  strings::NormalizeDigits(houseNumber);
+
+  if (!feature::IsHouseNumber(houseNumber))
     return false;
 
-  // Remove trailing zero's from house numbers.
-  // It's important for debug checks of serialized-deserialized feature.
-  string s(ss);
   uint64_t n;
-  if (strings::to_uint64(s, n))
-    s = strings::to_string(n);
+  if (strings::to_uint64(houseNumber, n))
```

Comment:
If you need to remove leading and trailing zeroes, you need to use Trim(), not to_uint64 + to_string.